azure-iot-device/azure/iot/device/iothub/config.py

```diff
@@ -0,0 +1,29 @@
+# -------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for
+# license information.
+# --------------------------------------------------------------------------
+
+import logging
+from azure.iot.device.common.config import BasePipelineConfig
+
+logger = logging.getLogger(__name__)
+
+
+class IoTHubPipelineConfig(BasePipelineConfig):
+    """A base class for storing all configurations/options shared across the Azure IoT Python Device Client Library.
+    More specific configurations such as those that only apply to the IoT Hub Client will be found in the respective
+    config files.
+    """
+
+    super
+
+    def __init__(self, product_info=None, **kwargs):
+        """Initializer for IoTHubPipelineConfig which passes all unrecognized keyword-args down to BasePipelineConfig
+        to be evaluated. This stacked options setting is to allow for unique configuration options to exist between the
+        IoTHub Client and the Provisioning Client, while maintaining a base configuration class with shared config options.
+
+        :param string product_info: A custom identification string for the type of device connecting to Azure IoT Hub.
+        """
+        super(IoTHubPipelineConfig, self).__init__(**kwargs)
+        self.product_info = product_info
```

Comment:
Why is this called `product_info` rather than `user_agent` or something

Reply:
User Agent defines the actual user agent that we are generating for the SDK. `custom_user_agent` could be a good name, but the powers that be have decided that it should be called Product Info. It is that way in C, `product_info`, and in node, `productInfo`.

---
In reply to: [337672739](https://github.com/Azure/azure-iot-sdk-python/pull/368#discussion_r337672739) [](ancestors = 337672739)